rust/lance/src/index/scalar.rs

```diff
@@ -256,6 +257,34 @@ impl IndexDetails {
             self.get_plugin().map(|p| p.version())
         }
     }
+
+    pub fn index_type(&self) -> IndexType {
```

Comment:
What is this method needed for?  I'd like to move away from these kinds of switch statements in indexes.